torch/_weights_only_unpickler.py

```diff
@@ -151,8 +198,107 @@ def load(self):
+                    # (1) The <module> that defines them is in `sys.modules`
+                    #     (we will use getattr_static to access it to ensure no code execution)
+                    # (2) They inherit from `torch.Tensor`
+                    # (2) The class is not overriding any of the `torch.Tensor` methods listed here:
+                    #     `__getattr__`, `__get__`, `__getattribute__`, `__setstate__`, `__set__`,
+                    #     and `tp_alloc`
+                    #     The methods that we ban overriding were selected in a test-driven manner
+                    #     by overriding every callable method on a tensor subclass and determinining
+                    #     which might get called when unpickling.
+                    # Further, these classes are only ever allowed as argument 2 of
+                    # `torch._tensor._rebuild_from_type_v2`.
+                    if module == "__builtin__":
+                        raise RuntimeError(f"Unsupported class {full_path}")
+                    elif module not in modules:
+                        # TODO: add a link here to a doc that explains to users what we mean by trust
+                        raise RuntimeError(
+                            f"Found GLOBAL `{full_path}` instruction in the pickle file but `{full_path}` was "
+                            f"not in the pre-defined list of allowed globals that are considered safe by the "
+                            "weights_only unpickler for rebuilding state_dicts. This is the expected behavior if "
+                            f"`{full_path}` is a user-defined tensor subclass not defined in the `torch` package. "
+                            f"If this is the case, we expect `{module}` to be present in `sys.modules` (i.e. it "
+                            "must be imported in the current environment), but this was not the case. "
+                            f"If you intend to unpickle a `{full_path}` please import `{name}` from `{module}`. "
+                            f"Note that having this imported will *only* allow the type `{full_path}` to be passed "
+                            "as the second argument to `torch._tensor._rebuild_from_type_v2`, which should enable "
+                            "the tensor subclass to be unpickled without any arbitrary code execution as long as "
+                            # If the user imports and these are overridden the next error will prompt them to use
+                            # torch.serialization.add_safe_globals.
+                            "a pre-defined list of methods called when unpickling are not overridden. In particular, "
+                            "the methods are `__getattr__`, `__get__`, `__getattribute__`, `__setstate__`, `__set__`, "
+                            "as well as the implementation of `tp_alloc`."
+                        )
+                    else:
+                        class_type = getattr_static(modules[module], name)
```

Comment:
Can we check what the error is if my module uses `__getattr__` to dynamically serve the object?
In case it is very obscure, we should catch it here and add context that we're using getattr_static and so things might be missing.

Reply:
Is this what you mean:

```python
from inspect import getattr_static
class SpoofModule(ModuleType):
    def __getattr__(self, name):
        if name == 'TestEmptySubclass':
            return TestEmptySubclass

spoof_mod = SpoofModule('bla')
print(getattr(spoof_mod, 'TestEmptySubclass'))  # <class '__main__.TestEmptySubclass'> 
print(getattr_static(spoof_mod, 'TestEmptySubclass'))  # AttributeError: TestEmptySubclass
```

I don't think this error is expressive enough that we can catch it 🤔  wdyt?

Reply:
I would say:
```
try:
  class_type = getattr_static(modules[module], name)
except AttributeError as e:
  raise AttributeError("Some context about getattr_state") from e
```